toolchain/check/context.cpp

```diff
@@ -618,6 +618,28 @@ auto Context::ParamOrArgEnd(Parse::NodeKind start_kind) -> SemIR::InstBlockId {
   return ParamOrArgPop();
 }
 
+auto Context::FinalizeInits() -> void {
+  inst_block_stack().PushInit();
+  if (!inst_block_stack().PeekCurrentBlockContents().empty()) {
```

Comment:
I did, but I just couldn't come up with a minimal way to do it.

The `GlobalInit` is being constructed inside the stack object, so there is no direct way to either "view" or "manipulate" it. I had to check if it was empty or not, and if it wasn't, I had to push it onto the stack again to add the `Return` instruction. I just thought I would push it regardless, that way I would be able to both "check its emptiness" and "add the return".

I realize it is not the most straight-forward, but I didn't want to make any more API changes to `InstBlockStack` than I had to.

Reply:
Makes sense, with the `return` function. Makes me wonder if we should switch around how we organize `Finalize` logic, now that we're adding more (if we had a single entry point for both exports and __global_init then it might look a little more reasonable to put this full function on InstBlockStack), but this seems good for now.